### PR TITLE
Handle HEAD requests for webhook validation

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

Some external systems need to do some webhook validation while setting up webhooks, e.g. https://developer.atlassian.com/cloud/trello/guides/rest-api/webhooks. This PR makes it so we repsond to `HEAD` requests to support this use case.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-48446

### CODE REVIEW

**Review by hanzei — APPROVED** [review]
Sounds OK to me. Let's make sure this special case get's documented.

File: server/httpin/service.go [hunk 1]
```diff
@@ -55,6 +55,8 @@ func NewService(proxy proxy.Service, appservices appservices.Service, conf confi
 	// Incoming remote webhooks.
 	h.HandleFunc(path.Webhook, h.Webhook).Methods(http.MethodPost)
 	h.HandleFunc(path.Webhook+"/{path}", h.Webhook).Methods(http.MethodPost)
+	h.HandleFunc(path.Webhook, h.WebhookValidate).Methods(http.MethodHead)
```
Comment (hanzei) [inline server/httpin/service.go:58]: Should there be a comment explaining why this special case exists?

**Review by hanzei — CHANGES_REQUESTED** [review]
Is it reasonable to write an restapi test to double check that the implementation works?

File: server/httpin/service.go (hunk 1 above)
Comment (levb) [reply server/httpin/service.go:58]: 1/5 seems pretty self-explanatory to me, That's a typical use case for HEAD. However, should we [erform the secret validation, and if so perhaps `s/webhookValidate/webhookValidateAuthentication/` to better reveal the intention?

File: server/httpin/webhook.go [hunk 2]
```diff
@@ -18,6 +18,10 @@ func (s *Service) Webhook(r *incoming.Request, w http.ResponseWriter, req *http.
 	}
 }
 
+func (s *Service) WebhookValidate(r *incoming.Request, w http.ResponseWriter, req *http.Request) {
+	w.WriteHeader(http.StatusOK)
```
Comment (levb) [inline server/httpin/webhook.go:22]: is w.WriteHeader necessary? I thought doing nothing is a 200 OK already.

**Review by levb — CHANGES_REQUESTED** [review]
1/5 I think the HEAD requests should be authenticated, and ATM this code is in `proxy.InvokeWebhook` so is not invoked for HEAD. We'd need to change it and validate HEAD also.

**Review by levb — APPROVED** [review]

**Comment by levb** [comment]
@mickmister do you mind submitting a (rough draft) docs PR for this? Just so that @asatkinson and the docs team have a clear todo item with the relevant info?

**Comment by mickmister** [comment]
@neflyte The App framework now supports webhook validation through HEAD requests, and our docs need to be updated to reflect this. This was a requirement for one of the systems we're integrating with (Trello). Please see the third paragraph under "Creating a Webhook" here https://developer.atlassian.com/cloud/trello/guides/rest-api/webhooks/#creating-a-webhook

**Review by hanzei — APPROVED** [review]

**Comment by mickmister** [comment]
Docs PR here https://github.com/mattermost/mattermost-developer-documentation/pull/1184

**Comment by javaguirre** [comment]
@mickmister I'm trying to test this, there was a hello-oauth2 example app, but I can't find it, I'll continue searching. How are you testing the HEAD call (I mean manually, I've checked the automated tests and those are great :-))? Thank you!

**Comment by levb** [comment]
The testing is covered with REST API test

**Comment by hanzei** [comment]
@mickmister Heads up that this change still needs to be documented

**Pull request merged**